src/unit_tests/method/method_order_test.cpp

```diff
@@ -0,0 +1,181 @@
+*   all copies or substantial portions of the Software.                             *
+*                                                                                   *
+*   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR      *
+*   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,        *
+*   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE     *
+*   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER          *
+*   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,   *
+*   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE   *
+*   SOFTWARE.                                                                       *
+*                                                                                   *
+*************************************************************************************/
+
+#include <rttr/registration>
+#include <catch/catch.hpp>
+#include <string>
+
+using namespace rttr;
+
+struct method_order_test_base
+{
+    method_order_test_base() = default ;
+    virtual ~method_order_test_base() = default ;
+
+    std::string whoami() { return "I am a base method" ; }
+    virtual std::string vwhoami() { return "I am a virtual base method" ; }
+
+    RTTR_ENABLE()
+
+};
+
+
+// RTTR_ENABLE incompatible with clang
+#define RTTR_ENABLE_OVERRIDE(...) \
+public:\
+    RTTR_INLINE rttr::type get_type() const override { return rttr::detail::get_type_from_instance(this); }  \
+    RTTR_INLINE void* get_ptr() override { return reinterpret_cast<void*>(this); } \
+    RTTR_INLINE rttr::detail::derived_info get_derived_info() override { return {reinterpret_cast<void*>(this), rttr::detail::get_type_from_instance(this)}; } \
+    using base_class_list = TYPE_LIST(__VA_ARGS__);
+
+
+struct method_order_test_derived : method_order_test_base
+{
+    method_order_test_derived() = default ;
+    ~method_order_test_derived() override = default ;
+
+    std::string whoami() { return "I am a non-virtual derived method with same name" ; }
+    std::string vwhoami() override { return "I am a virtual derived method with same name" ; }
+
+    RTTR_ENABLE_OVERRIDE(method_order_test_base)
+};
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+RTTR_REGISTRATION
+{
+    registration::class_<method_order_test_base>("method_order_test_base")
+        .method("whoami", &method_order_test_base::whoami)
+        .method("vwhoami", &method_order_test_base::vwhoami)
+        ;
+    registration::class_<method_order_test_derived>("method_order_test_derived")
+        .method("whoami", &method_order_test_derived::whoami)
+        .method("vwhoami", &method_order_test_derived::vwhoami)
+        ;
+}
+
+// approach 1 looks up the "method instance" without considering argument type and invoke the registered function()
+template<class TT>
+std::string rttr_invoke_approach1(
```

Comment:
No need for a function template. 
Try this:
```
std::string rttr_invoke_approach1(rttr::instance inst, rttr::string_view meth_name);
```

PS: All braces, `{`, should be placed in the same column, otherwise the CI tests will not pass. Thats the coding style for RTTR. Check travis, first build.

Reply:
done - prefer const rttr::instance& to templaet